pkg/agent/baker_test.go

```diff
@@ -0,0 +1,180 @@
+					ContainerRegistryDNSSuffix:   "testContainerRegistryDNSSuffix",
+					CosmosDBDNSSuffix:            "testCosmosDBDNSSuffix",
+					TokenAudience:                "testTokenAudience",
+				},
+			},
+		}
+		cs.Properties.LinuxProfile.SSH.PublicKeys = []api.PublicKey{{
+			KeyData: string("testsshkey"),
+		}}
+
+		// AKS always pass in te customHyperKubeImage to aks-e, so we don't really rely on
+		// the default component version for "hyperkube", which is not set since 1.17
+		if IsKubernetesVersionGe(k8sVersion, "1.17.0") {
+			cs.Properties.OrchestratorProfile.KubernetesConfig.CustomHyperkubeImage = fmt.Sprintf("k8s.gcr.io/hyperkube-amd64:v%v", k8sVersion)
+		}
+
+		if containerServiceUpdator != nil {
+			containerServiceUpdator(cs)
+		}
+
+		agentPool := cs.Properties.AgentPoolProfiles[0]
+		baker := InitializeTemplateGenerator()
+
+		// customData
+		customData := baker.GetNodeBootstrappingPayload(cs, agentPool)
+		// Uncomment below line to generate test data in local if agentbaker is changed in generating customData
+		// backfillCustomData(folder, customData)
+		expectedCustomData, err := ioutil.ReadFile(fmt.Sprintf("./testdata/%s/CustomData", folder))
+		if err != nil {
+			panic(err)
+		}
+		Expect(customData).To(Equal(string(expectedCustomData)))
+
+		// CSE
+		cseCommand := baker.GetNodeBootstrappingCmd(cs, agentPool,
+			"tenantID", "subID", "resourceGroupName", "userAssignedID", true, true)
+		// Uncomment below line to generate test data in local if agentbaker is changed in generating customData
+		// ioutil.WriteFile(fmt.Sprintf("./testdata/%s/CSECommand", folder), []byte(cseCommand), 0644)
+		expectedCSECommand, err := ioutil.ReadFile(fmt.Sprintf("./testdata/%s/CSECommand", folder))
+		if err != nil {
+			panic(err)
+		}
+		Expect(cseCommand).To(Equal(string(expectedCSECommand)))
+
+	}, Entry("AKSUbuntu1604 with k8s version less than 1.18", "AKSUbuntu1604+K8S115", "1.15.7", nil),
+		Entry("AKSUbuntu1604 with k8s version 1.18", "AKSUbuntu1604+K8S118", "1.18.2", nil),
+		Entry("AKSUbuntu1604 with k8s version 1.17", "AKSUbuntu1604+K8S117", "1.17.7", nil),
+		Entry("AKSUbuntu1604 with Temp Disk", "AKSUbuntu1604+TempDisk", "1.15.7", func(cs *api.ContainerService) {
+			cs.Properties.OrchestratorProfile.KubernetesConfig.ContainerRuntimeConfig = map[string]string{
+				common.ContainerDataDirKey: "/mnt/containers",
+			}
+		}),
+		Entry("AKSUbuntu1604 with Temp Disk and containerd", "AKSUbuntu1604+TempDisk+Containerd", "1.15.7", func(cs *api.ContainerService) {
+			cs.Properties.OrchestratorProfile.KubernetesConfig.ContainerRuntimeConfig = map[string]string{
+				common.ContainerDataDirKey: "/mnt/containers",
+			}
+			cs.Properties.AgentPoolProfiles[0].KubernetesConfig.ContainerRuntime = api.Containerd
+		}),
+		Entry("AKSUbuntu1604 with RawUbuntu", "RawUbuntu", "1.15.7", func(cs *api.ContainerService) {
+			cs.Properties.AgentPoolProfiles[0].Distro = api.Ubuntu
+		}),
```

Comment:
I don't see a test case for dynamic kubelet enabled scenario?

Reply:
will enable afterwards